connector/src/main/java/org/jboss/as/connector/subsystems/datasources/GetInstalledDriverOperationHandler.java

```diff
@@ -72,12 +73,24 @@ public void execute(final OperationContext context, final ModelNode operation) t
                         driverNode.get(DRIVER_XA_DATASOURCE_CLASS_NAME.getName());
                     } else {
                         driverNode.get(DEPLOYMENT_NAME.getName());
-                        driverNode.get(DRIVER_MODULE_NAME.getName()).set(driver.getModuleName().getName());
-                        driverNode.get(MODULE_SLOT.getName()).set(driver.getModuleName() != null ? driver.getModuleName().getSlot() : "");
+                        String moduleName = driver.getModuleName();
+                        // Retrieve the canonical module identifier string using ModuleIdentifierUtil
+                        String canonicalModId = ModuleIdentifierUtil.canonicalModuleIdentifier(moduleName);
+                        // Check if the canonicalModId contains a colon (indicating it's a valid "module:slot" format)
+                        String slot = "";
+                        if (canonicalModId != null && canonicalModId.contains(":")) {
+                            // Extract the slot by splitting the canonicalModId
+                            String[] parts = canonicalModId.split(":");
+                            if (parts.length > 1) {
+                                slot = parts[1];
+                            }
+                        }
+                        driverNode.get(DRIVER_MODULE_NAME.getName()).set(moduleName);
+                        driverNode.get(MODULE_SLOT.getName()).set(slot);
```

Comment:
I made this same comment on #18560...

I was thinking of consolidating this kind of thing in ModuleIdentifierUtil:

https://github.com/wildfly/wildfly-core/pull/6282/files#diff-867089bef7659adb3ead7e82d208a38aa95d3fda6df7476187f2aa272bf064d4R26-R41

So this could just be another function to pass in.

Reply:
@bstansberry should we wait for the PR to get merged?

Reply:
Yes, I think so.

The 'name' part can include ':' if it is escaped. "foo\:bar:baz" is valid.

Reply:
https://github.com/wildfly/wildfly-core/pull/6293